src/app/test-results/test-results-list/test-results-list.component.html

```diff
@@ -0,0 +1,50 @@
+<div id="test-results-list">
+  <p-table [value]="summaries">
+    <ng-template pTemplate="header">
+      <tr>
+        <th class="col-id">ID</th>
+        <th class="col-name">FILENAME</th>
+        <th class="col-status">STATUS</th>
+      </tr>
+    </ng-template>
+    <ng-template pTemplate="body" let-summary>
+      <tr role="button" (click)="this.toggleDetails(summary)">
+        <td class="col-id">{{summary.testResultSummaryId}}</td>
+        <td class="col-name">{{summary.fileName}}</td>
+        <td class="col-status">{{summary.failed ? "Failed" : "Passed"}}</td>
+      </tr>
+      <tr *ngIf="summary.testResultSummaryId === this.selectedSummaryId">
+        <td colspan="3">
+          <div>
+            <a *ngIf="this.failedDetails?.length === 0">No messages</a>
+            <table class="detail-list" *ngIf="this.failedDetails?.length > 0">
+              <tr>
+                <th class="col-test-name">Name</th>
+                <th class="col-test-status">Status</th>
+              </tr>
+              <tr>
+                <td colspan="2">
+                  <div>
+                    <table *ngFor="let detail of this.failedDetails">
+                      <tr role="button" (click)="this.showDetailMessage(detail)">
```

Comment:
3 nested tables. Suggest styling and aligning with divs instead.

Tables are awesome to get things aligned, but as the 2 deeper tables only have 2 rows each where the 2nd row has `colspan="2"`, a group of divs would solve it much easier I think.

Perhaps try to use the PrimeNG's accordion for this reveal feature you're using as well https://www.primefaces.org/primeng/showcase/#/accordion

Reply:
Maybe use a `p-tree` instead as well? https://www.primefaces.org/primeng/showcase/#/tree

Reply:
I have seen https://www.primefaces.org/primeng/showcase/#/table/rowexpansion work really well for these nested scenarios too when you want details from a list.